client/my-sites/checkout/src/hooks/use-tos-foldable-card.tsx

```diff
@@ -7,12 +7,16 @@ import { getSelectedSiteId, getSectionName } from 'calypso/state/ui/selectors';
 
 export function useToSFoldableCard(): boolean {
 	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
-
+	const section = useSelector( getSectionName );
 	const isJetpackNotAtomic = useSelector(
 		( state ) => isJetpackSite( state, siteId ) && ! isSiteAutomatedTransfer( state, siteId )
 	);
+
+	const enabledSections = [ 'business-plan-upgrade-upsell', 'checkout' ];
+
 	const isWPcomCheckout =
-		useSelector( getSectionName ) === 'checkout' &&
+		section != null &&
```

Comment:
Since `getSectionName` returns `null` by default, a truthy check should be enough.

```suggestion
		section && enabledSections.includes( section ) &&
```

Also just for readability, I would group the `section` and `enabledSections` definitions together (instead of having `isJetpackNotAtomic` defined in between.